packages/atlas-service/src/components/ai-signin-modal.tsx

```diff
@@ -57,68 +64,71 @@ const AISignInModal: React.FunctionComponent<SignInModalProps> = ({
   onSignInModalClose,
   onSignInClick,
 }) => {
+  const darkMode = useDarkMode();
+
   return (
-    <Modal
+    <MarketingModal
+      darkMode={darkMode}
+      graphic={<AISignInImageBanner></AISignInImageBanner>}
+      // @ts-expect-error leafygreen only allows strings, but we need to pass
+      // badge component
+      title={
+        <div className={cx(titleStyles, darkMode && titleDarkModeStyles)}>
+          Build faster with AI&nbsp;
+          <Badge className={badgeStyles} variant="blue">
+            Experimental
+          </Badge>
+        </div>
+      }
       open={isSignInModalVisible}
-      setOpen={(open) => {
-        if (open === false) {
-          onSignInModalClose?.();
+      onClose={onSignInModalClose}
+      // @ts-expect-error leafygreen only allows strings, but we need to pass
+      // icons
+      buttonText={
+        <>
+          <Icon glyph="LogIn"></Icon>&nbsp;Log in to Atlas to enable
+          {isSignInInProgress && (
+            <>
+              &nbsp;
+              <SpinLoader
+                // Marketing modal button is always bright, spin loader in dark mode gets lost
+                darkMode={false}
+              ></SpinLoader>
+            </>
+          )}
+        </>
+      }
+      onButtonClick={() => {
+        // We can't control buttons in marketing modal, so instead we just do
+        // nothing when button is clicked and sign in is in progress
+        if (isSignInInProgress) {
```

Comment:
Double checking, does this make it so that a new browser window does not open if the user clicks the sign in button again? If the user accidentally closed that browser window does that make `isSignInInProgress` `false`?

Reply:
Only one sign in process is guaranteed by oidc-plugin and atlas-service implementation, this is just so that we don't dispatch unnecessary redux action in this case. Closing a browser window before finishing the auth flow will not throw an error, oidc-plugin has no means to tracking whether or not someone just closed the tab, you'd need to close the modal in compass to interrupt it